packages/react/select/src/Select.tsx

```diff
@@ -839,18 +747,284 @@ const SelectContentImpl = React.forwardRef<SelectContentImplElement, SelectConte
+    const [contentZIndex, setContentZIndex] = React.useState<string>();
+    useLayoutEffect(() => {
+      if (content) setContentZIndex(window.getComputedStyle(content).zIndex);
+    }, [content]);
+
+    // When the viewport becomes scrollable at the top, the scroll up button will mount.
+    // Because it is part of the normal flow, it will push down the viewport, thus throwing our
+    // trigger => selectedItem alignment off by the amount the viewport was pushed down.
+    // We wait for this to happen and then re-run the positining logic one more time to account for it.
+    const handleScrollButtonChange = React.useCallback(
+      (node: SelectScrollButtonImplElement | null) => {
+        if (node && shouldRepositionRef.current === true) {
+          position();
+          focusSelectedItem?.();
+          shouldRepositionRef.current = false;
+        }
+      },
+      [position, focusSelectedItem]
+    );
+
+    return (
+      <SelectViewportProvider
+        scope={__scopeSelect}
+        contentWrapper={contentWrapper}
+        shouldExpandOnScrollRef={shouldExpandOnScrollRef}
+        onScrollButtonChange={handleScrollButtonChange}
+      >
+        <div
+          ref={setContentWrapper}
+          style={{
+            display: 'flex',
+            flexDirection: 'column',
+            position: 'fixed',
+            zIndex: contentZIndex,
+          }}
+        >
+          <Primitive.div
+            {...popperProps}
+            ref={composedRefs}
+            style={{
+              // When we get the height of the content, it includes borders. If we were to set
+              // the height without having `boxSizing: 'border-box'` it would be too big.
+              boxSizing: 'border-box',
+              // We need to ensure the content doesn't get taller than the wrapper
+              maxHeight: '100%',
+              ...popperProps.style,
+            }}
+          />
+        </div>
+      </SelectViewportProvider>
+    );
+  }
+);
+
+SelectPopperAbove.displayName = POPPER_ABOVE_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * SelectPopperOutside
+ * -----------------------------------------------------------------------------------------------*/
+
+const POPPER_OUTSIDE_NAME = 'SelectPopperOutside';
+
+type SelectPopperOutsideElement = React.ElementRef<typeof PopperPrimitive.Content>;
+type PopperContentProps = React.ComponentPropsWithoutRef<typeof PopperPrimitive.Content>;
+interface SelectPopperOutsideProps extends PopperContentProps, SelectPopperPrivateProps {}
+
+const SelectPopperOutside = React.forwardRef<SelectPopperOutsideElement, SelectPopperOutsideProps>(
+  ({ __scopeSelect, ...props }: ScopedProps<SelectPopperOutsideProps>, forwardedRef) => {
+    const popperScope = usePopperScope(__scopeSelect);
+    const context = useSelectContext(CONTENT_NAME, __scopeSelect);
+    const {
+      align = context.dir === 'rtl' ? 'end' : 'start',
```

Comment:
Related https://github.com/radix-ui/primitives/pull/1881#discussion_r1068206714, we should discuss leaning into the floating ui implementation for handling this, I think this appears to applying correctly in the current stories only because they are using `Portal`.